lib/did_you_mean/finders/name_error_finders/similar_class_finder.rb

```diff
@@ -32,6 +33,13 @@ def scopes
 
     private
 
+    def autoload_class_name_inflector
+      return unless defined?(Rails)
+      name_is_singular = name_from_message.to_s.pluralize.singularize == name_from_message.to_s
+      class_name_inflector = name_is_singular ? name_from_message.to_s.pluralize : name_from_message.to_s.singularize
+      class_name_inflector.safe_constantize
+    end
+
```

Comment:
Anyone can override `#const_missing` or do something like `if defined?(Model)` for development purposes. If they assume that constants are auto-loaded by Rails, then `did_you_mean` would break it. In other words, people who assume so will be confused by the side effect. In addition, [Configuring `ActiveSupport::Inflector::Inflections`](http://api.rubyonrails.org/classes/ActiveSupport/Inflector/Inflections.html) could affect the code here.

Also, `#autoload_class_name_inflector ` is too specific as I haven't seen plural/singular class names both together. So I don't think it really helps many people.